pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -163,6 +165,11 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 		for _, splitReq := range splitReqs {
 			// Do not try to pick response from cache at all if the request is not cachable.
 			if cachable, reason := isRequestCachable(splitReq.orig, maxCacheTime, cacheUnalignedRequests, s.logger); !cachable {
+				level.Debug(s.logger).Log("msg", "skipping response cache as query is not cacheable", "query", splitReq.orig.GetQuery(), "reason", reason, "tenants", tenant.JoinTenantIDs(tenantIDs))
```

Comment:
Sorry I wasn't clear. The code here should use `spanLog` now like it's a regular logger and only call it once. E.g. both these logging calls would be replaced by:
```go
level.Debug(spanLog).Log(<all the kv pairs>)
```

Reply:
😅 sorry about that. it should be better now.